### ISSUE TITLE
Option to sign off on unsupported constructs

### ISSUE THREAD

**Issue by zhassan-aws** [model-checking/kani#1097]
Requested feature: Currently, when a crate contains an unsupported function (e.g. local asm), Kani codegens it as assert/assume false to cause verification to fail if that construct is reachable. If it is reachable, only failures are reported, and non-failures are reported as "UNDETERMINED".

In some cases, the user might want to override this behavior and allow verification to continue despite the reachability of the unsupported construct, e.g. if they're confident the construct would not impact the results of other properties. Thus, we should add an option and/or a Config.toml setting to allow the user to either globally override this behavior or allow a more fine-grained control (e.g. at the crate level).

When the option is enabled, Kani would:
1. Not add the assume(false) in the codegen of the unsupported construct (and perhaps not inject an assert(false) in the first place)
2. Not change "SUCCESS" to "UNDETERMINED" if an unsupported construct is found to be reachable.

Use case:
Link to relevant documentation (Rust reference, Nomicon, RFC):
Is this a breaking change?

Test case:

```rust

#![feature(asm)]
fn unsupp(x: &mut u8) {
    unsafe {
        std::arch::asm!("nop");
    }
}

#[kani::proof]
fn main() {
    let mut x = 0;
    unsupp(&mut x);
    assert!(x == 0);
}
```
